src/tokenizer/ngram_tokenizer.rs

```diff
@@ -100,6 +109,7 @@ impl<'a> NgramTokenStream<'a> {
     /// returning None if processing should stop
     fn chomp(&mut self) -> Option<(usize, usize)> {
```

Comment:
The `start`, `len` is ok semantics, but it does interface well with rust slicing. 
 `start_offset`, `stop_offset` could have been better.

It is important to document the meaning of the returned value in that kind of function, especially when it is ambiguous like this. Even if it is private.